internal/services/grpc/middleware/auth.go

```diff
@@ -41,6 +43,10 @@ func (a *GRPCAuthN) Middleware(ctx context.Context) (context.Context, error) {
 		return nil, forbidden
 	}
 
+	// we want to rate limit by API token
+
+	ctx = context.WithValue(ctx, "rate_limit_token", hashToken(token))
```

Comment:
The token's ID is encoded in the token and should be relatively easy to return from the token package (similar to how we're returning the tenant id). I'd feel more comfortable from a security perspective using the UUID corresponding to the token than setting a hash of the actual token value in the context.

Reply:
I now use the ID of the token from the db.